data-serving/data-service/api/openapi.yaml

```diff
@@ -221,6 +258,23 @@ components:
           type: array
           items:
             $ref: "#/components/schemas/Case"
+    NewCase:
+      description: A "#/components/schemas/Case" with additional required fields for newly-created cases
+      allOf:
+        - $ref: "#/components/schemas/Case"
+      properties:
+        caseReference:
+          required:
+            - sourceId
+            - sourceUrl
+        location:
+          required:
```

Comment:
This is another area in which there's a conditional requirement. If lat/long aren't provided, then `query` is required in order to geocode. So there's this sort of `required: $or: { $and: { lat, lng } }, query}` relationship going on.

WDYT is the best way to handle this?

Reply:
I think that applies to the curator service, not data service -- that said, 2 things:

1. When I start doing the API for the curator service, we'll run into this. My proposal is to add a request interceptor for create (and maybe upsert?) that runs the more complex validation logic that we can't express in the OpenAPI spec, and also document these requirements in the `description` field for the property in the OpenAPI spec.

2. You made me realize I did kinda mess up location here though -- s/query/name and missed `geometry` completely. I've updated the location spec

Reply:
Oof -- completely glossed that this was for the data service, sorry. Thanks for the catch and SGTM.